Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -314,23 +324,46 @@ void OpenALStream::SoundLoop()
     // many silence samples.  These do not need to be timestretched.
     if (rate > 0.10)
     {
-      soundTouch.setTempo(rate);
-      if (rate > 10)
+      if (use_timestretching)
+      {
+        soundTouch.setTempo(rate);
+      }
+      else
+      {
+        alSourcef(ui_source, AL_PITCH, static_cast<ALfloat>(rate));
```

Comment:
This still doesn't make sense to me. Time-stretching and pitch-shifting are the same thing—if a user disables time-stretching, then it shouldn't just switch to a different implementation.

Does OpenAL actually do pitch-shifting, or does only resample? If it's resampling, the Mixer class already does that internally.